go/pkg/sysdb/metastore/db/dao/task.go

```diff
@@ -58,6 +60,55 @@ func (s *taskDb) GetByName(inputCollectionID string, taskName string) (*dbmodel.
 	return &task, nil
 }
 
+func (s *taskDb) GetByID(taskID uuid.UUID) (*dbmodel.Task, error) {
+	var task dbmodel.Task
+	err := s.db.
+		Where("task_id = ?", taskID).
+		Where("is_deleted = ?", false).
+		First(&task).Error
+
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, nil
+		}
+		log.Error("GetByID failed", zap.Error(err), zap.String("task_id", taskID.String()))
+		return nil, err
+	}
+	return &task, nil
+}
+
+func (s *taskDb) AdvanceTask(taskID uuid.UUID, taskRunNonce uuid.UUID) error {
```

Comment:
FYI I'm going to remove/modify this later. The path that updates next_nonce also needs to transactionally update completion_offset.